app/models/concerns/orchestration/dns.rb

```diff
@@ -1,175 +1,211 @@
-    return true unless dns? || reverse_dns?
+    logger.info "IPv4 DNS not supported for #{name}, skipping orchestration rebuild" unless dns?
+    logger.info "IPv6 DNS not supported for #{name}, skipping orchestration rebuild" unless dns6?
+    logger.info "Reverse IPv4 DNS not supported for #{name}, skipping orchestration rebuild" unless reverse_dns?
+    logger.info "Reverse IPv6 DNS not supported for #{name}, skipping orchestration rebuild" unless reverse_dns6?
+    return true unless dns? || dns6? || reverse_dns? || reverse_dns6?
     del_dns_a_record_safe
+    del_dns_aaaa_record_safe
     del_dns_ptr_record_safe
-    a_record_result, ptr_record_result = true, true
+    del_dns_ptr6_record_safe
+    a_record_result = true
+    aaaa_record_result = true
+    ptr_record_result = true
+    ptr6_record_result = true
     begin
       a_record_result = recreate_a_record if dns?
+      aaaa_record_result = recreate_aaaa_record if dns6?
       ptr_record_result = recreate_ptr_record if reverse_dns?
-      a_record_result && ptr_record_result
+      ptr6_record_result = recreate_ptr6_record if reverse_dns6?
+      a_record_result && aaaa_record_result && ptr_record_result && ptr6_record_result
     rescue => e
-      Foreman::Logging.exception "Failed to rebuild DNS record for #{name}, #{ip}", e, :level => :error
+      Foreman::Logging.exception "Failed to rebuild DNS record for #{name}", e, :level => :error
       false
     end
   end
 
   def dns_a_record
-    return unless dns? or @dns_a_record
-    @dns_a_record ||= Net::DNS::ARecord.new dns_record_attrs
+    return unless dns? || @dns_a_record
+    return unless ip_available?
+    @dns_a_record ||= Net::DNS::ARecord.new dns_a_record_attrs
+  end
+
+  def dns_aaaa_record
+    return unless dns6? || @dns_aaaa_record
+    return unless ip6_available?
+    @dns_aaaa_record ||= Net::DNS::AAAARecord.new dns_aaaa_record_attrs
   end
 
   def dns_ptr_record
-    return unless reverse_dns? or @dns_ptr_record
+    return unless reverse_dns? || @dns_ptr_record
     @dns_ptr_record ||= Net::DNS::PTR4Record.new reverse_dns_record_attrs
   end
 
-  def del_dns_a_record_safe
-    if dns_a_record
-      begin
-        del_dns_a_record
-      rescue => e
-        Foreman::Logging.exception "Proxy failed to delete DNS a_record for #{name}, #{ip}", e, :level => :error
-      end
-    end
+  def dns_ptr6_record
+    return unless reverse_dns6? || @dns_ptr6_record
+    @dns_ptr6_record ||= Net::DNS::PTR6Record.new reverse_dns6_record_attrs
   end
 
-  def del_dns_ptr_record_safe
-    if dns_ptr_record
-      begin
-        del_dns_ptr_record
-      rescue => e
-        Foreman::Logging.exception "Proxy failed to delete DNS ptr_record for #{name}, #{ip}", e, :level => :error
+  RECORD_TYPES.each do |record_type|
+    define_method("del_dns_#{record_type}_record_safe") do # def del_dns_a_record_safe
+      if send(:"dns_#{record_type}_record") # dns_a_record
```

Comment:
These methods used to be protected, is there any reason to make them public now?